### PR TITLE
hackage candidates proposal

### PR DESCRIPTION
**Pull request by gbaz**
this is just the proposal from last year again. @hvr what do you think?

### CODE REVIEW

File: content/ideas/hackage-candidate.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: Finish the package candidate workflow for Hackage
+---
+
+Hackage candidate packages currently cannot be used directly, and their UI could be improved. We would like to have new packages be uploaded as candidates by default, to improve the vetting process. But this means polishing off candidate functionality. The main issues left to do are tracked [here](https://github.com/haskell/hackage-server/projects/1)
```
Comment (gwils) [inline content/ideas/hackage-candidate.md]: > We would like to have new packages be uploaded as candidates by default, to improve the vetting process.
Isn't this already the case in modern cabal versions?

File: content/ideas/hackage-candidate.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+The [first step](https://github.com/haskell/hackage-server/issues/635) is moving the candidate display page to the new templating system and sharing code with the main package page. Following this, we need to implement a [new candidate index](https://github.com/haskell/hackage-server/issues/597), able to be provided as a secondary index. This would be a "v1" index, and mutable.
+
+Beyond this we want to extend the docbuilder and docuploads to work with candidates, and then implement a fixed workflow from candidacy to validation and then publishing.
```
Comment (gwils) [inline content/ideas/hackage-candidate.md:9]: Is this a required part of this project or a stretch goal?

File: content/ideas/hackage-candidate.md (hunk 1 above)
Comment (maoe) [reply content/ideas/hackage-candidate.md]: IIRC if you upload a package from the web UI, it'll be published directly to the main index.

File: content/ideas/hackage-candidate.md (hunk 2 above)
Comment (gbaz) [reply content/ideas/hackage-candidate.md:9]: I'd say that as much or as little can be proposed as someone thinks reasonable. It's an idea. Students can propose how much they think they can handle.

File: content/ideas/hackage-candidate.md (hunk 1 above)
Comment (gbaz) [reply content/ideas/hackage-candidate.md]: I added some text to clear this up.

**Pull request merged**